dataeng/resources/detect-new-dbt-manual-models-fields.sh

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+set -ex
+
+# Setup a virtual environment
+PYTHON38_VENV="py38_venv"
+virtualenv --python=python3.8 --clear "${PYTHON38_VENV}"
+source "${PYTHON38_VENV}/bin/activate"
+
+# Go into analytics-tools and install the dependencies
+cd ${WORKSPACE}/analytics-tools/snowflake
+make requirements
+
+# I am not able to get make upgrade to work after adding pyyaml to base.in.
+# So, I am going to install pyyaml manually.
+pip install pyyaml
+
+# The extra vars file for this job contains both field mappings and the necessary credentials for Snowflake and Jenkins.
+# Therefore, the options to the script are read from the config file.
+CONFIG_PATH=${WORKSPACE}/analytics-secure/job-configs/DETECT_NEW_DBT_MANUAL_MODELS_FIELDS_JOB_EXTRA_VARS.yaml
+# Invoke the script to detect new fields that need to be added manually
+python detect_new_raw_columns.py ${CONFIG_PATH}
```

Comment:
You may have to change the parameters of this script to take snowflake credentials piecemeal, such as in most of our other scripts, e.g.: https://github.com/edx/analytics-tools/blob/6d9cdfa18a6a67d33a976f3baf38bb937526c7a0/snowflake/retirement_cleanup.py#L126-L144

Reply:
Also, job-configs yml files are really only meant to be consumed by the jenkins seed job, not anything else.  I think it could be confusing to future DE members reading this code as-is.